### PR TITLE
Add expectNext(d: FiniteDuration) and requestNext(d: FiniteDuration)

### PR DESCRIPTION
**Pull request by kbrowder**
For issue #20815, ran validatePullRequest and all tests passed.

### CODE REVIEW

**Comment by ktoso** [comment]
OK TO TEST

**Comment by 2beaucoup** [comment]
`requestNextWithin` / `expectNextWithin`?

File: akka-stream-testkit/src/main/scala/akka/stream/testkit/StreamTestKit.scala [hunk 1]
```diff
@@ -252,7 +252,14 @@ object TestSubscriber {
...
      */
     def expectNext(): I = {
-      val t = probe.remainingOr(probe.testKitSettings.SingleExpectDefaultTimeout.dilated)
+      expectNext(probe.testKitSettings.SingleExpectDefaultTimeout.dilated)
+    }
+
+    /**
+     * Expect and return a stream element during specified time or timeout.
+     */
+    def expectNext(d: FiniteDuration): I = {
```
Comment (patriknw) [inline akka-stream-testkit/src/main/scala/akka/stream/testkit/StreamTestKit.scala]: I think the parameter is named `max` in other places

Comment (patriknw) [inline akka-stream-testkit/src/main/scala/akka/stream/testkit/StreamTestKit.scala]: ok, seems to be inconsistent, and `d` is used for the other `expectNext` so it's alright

**Comment by patriknw** [comment]
LGTM

File: akka-stream-testkit/src/main/scala/akka/stream/testkit/StreamTestKit.scala [hunk 2]
```diff
@@ -613,6 +620,11 @@ object TestSubscriber {
       subscription.request(1)
       expectNext()
     }
+
+    def requestNext(d: FiniteDuration): T = {
```
Comment (ktoso) [inline akka-stream-testkit/src/main/scala/akka/stream/testkit/StreamTestKit.scala]: could you add the comment here as well?

```
+    /**
 +     * Expect and return a stream element during specified time or timeout.
 +     */
```

**Comment by ktoso** [comment]
LGTM

File: akka-stream-testkit/src/main/scala/akka/stream/testkit/StreamTestKit.scala (hunk 2 above)
Comment (kbrowder) [inline akka-stream-testkit/src/main/scala/akka/stream/testkit/StreamTestKit.scala]: OK, comments to the various flavors or requestNext incomming

**Comment by ktoso** [comment]
Thanks a lot :-)

**Comment by kbrowder** [comment]
Thanks for merging :D

**Pull request merged**